### DISCUSSION TITLE
Feedback: Streaming

### DISCUSSION THREAD

**Discussion by JasperV** [mswjs/mswjs.io#433] [Q&A] [answered]
- [Open page](https://mswjs.io/docs/recipes/streaming)

## Feedback

I am looking for an example of streaming that does Server Sent Event. I would love to mock those in my current project.

**Comment by kettanaito** [comment, marked as answer]
Hi, @JasperV 👋 

You can find a Server Sent Events example in here: https://alexocallaghan.com/mock-sse-with-msw. 

For posterity, here's the usage example:

```ts
import { http, HttpResponse } from "msw";

const encoder = new TextEncoder();

const handler = http.get('http://example.com/stream', () => {
  const stream = new ReadableStream({
    start(controller) {
      controller.enqueue(
        encoder(
          `event:some-event\ndata:some data\n\n`
        )
      );
      controller.close();
    },
  });

  return new HttpResponse(stream, {
    headers: {
      "Content-Type": "text/event-stream",
    }.
  });
});
```

I am also working on the first-class support for SSE events in https://github.com/mswjs/msw/pull/2299. It's already in a usable state. Let me know if you'd like to give it a try, I can do a preview release for you to install it in your project.

**Reply by kettanaito** [reply]
I won't add this example because it's too low level. I would rather document the new SSE support in MSW once it's out. For now, utilizing `ReadableStream` and sending SSE is considered implied knowledge.

**Reply by JasperV** [reply]
Hello @kettanaito,

I got it working with this example, thanks!

Sure, if it can be made easier, I would like to try that out. Does it come with a manual of some sort?

**Reply by kettanaito** [reply]
@JasperV I will document it in the upcoming weeks and publish a preview release. Just subscribe to the updates on the repo I linked and you won't miss it.
